### ISSUE TITLE
Evaluate expressions passed header, footer, and view's styles

### ISSUE THREAD

**Issue by amin-nas** [EnsembleUI/ensemble#847]
I want to be able to set properties dynamically across:

1. View style props (e.g. backgroundColor, scrollableView)
2. Header props
3. Footer props

```yaml
View:
  styles:
    backgroundColor: '${flag == true ? "green" : "red"}'
  header:
    title: Home
    styles:
      backgroundColor: '${flag == true ? "green" : "red"}'
  body:
    Column:
      styles:
        backgroundColor: '${flag == true ? "green" : "red"}'
      children:
        - Content
  footer:
    styles:
      backgroundColor: '${flag == true ? "green" : "red"}'
    children:
      - Content

Global: |-
  //@code
  var flag = true;

Content:
  body:
    Spacer:
      styles:
        size: 100
```

**Comment by kmahmood74** [comment]
this doesn't work for title as well. For header/title, the expression is evaluated only once but never updated. The whole header expression code is unfinished or just super buggy. 

@vusters @vinothvino42 

Example - 

```
 header:
    
    title: Welcome ${ensemble.storage.players[ensemble.storage.currentPlayer].name}
```

will work only once. Afterwards even if I change the currentIndex or players array and assign, it won't update. Same code will update fine if it was on a Text instead of title. Super buggy!

**Comment by kmahmood74** [comment]
appreciate if we start fixing these things

**Comment by kmahmood74** [comment]
see the use case in #763 as well

**Comment by hemish11** [comment]
Yes, https://github.com/EnsembleUI/ensemble/issues/763 is a duplicate of this, and for column and other widgets, this does work, it's only breaking for the View.

**Comment by kmahmood74** [comment]
@hemish11 this is not right. You missed the styles under View even though both #763 and #847 mentioned View specifically. I see this code - 

```
    LinearGradient? backgroundGradient = Utils.getBackgroundGradient(
        widget._pageModel.pageStyles?['backgroundGradient']);
    Color? backgroundColor =
        Utils.getColor(widget._pageModel.pageStyles?['backgroundColor']);
        ```
in Page.dart so it is clear that backgroundColor or gradient are not being evaluated. 

Please don't just pick and choose and do this for ALL STYLES on View.

**Comment by kmahmood74** [comment]
@hemish11 please do this at your earliest.

**Comment by kmahmood74** [comment]
@hemish11 I checked in the fix for the backgroundColor on View. Please evaluate expressions for other properties as well
